### PR TITLE
Intersection, Sort, and Subtract passing tests

### PR DESCRIPTION
**Pull request by TrevorJamesH**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
-[divide](Docs/divide.md)   
-[map](Docs/map.md)      
-[pick](Docs/pick.md)     
-[reject](Docs/reject.md)  
-[assoc](Docs/assoc.md)  
-[flatten](Docs/flatten.md)  
-[memoize](Docs/memoize.md)  
-[range](Docs/range.md)    
-[sort](Docs/sort.md)
+Description
```
Comment (Moniarchy) [inline README.md:1]: It isn't easily apparent to me what your specs are. Writing them out with markdown checkboxes in a bulleted list will be helpful once project reviews come around next Monday.

File: src/lib/intersection.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const intersection = (array1,array2) => {
```
Comment (Moniarchy) [inline src/lib/intersection.js:1]: this method still needs a test before this branch is merged. also, consider doing one method and its test for each branch, and naming the branch for the method you wrote.

File: src/lib/sort.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const sort = (startingArray) => {
```
Comment (Moniarchy) [inline src/lib/sort.js:1]: see above re: testing and standalone branches

File: src/lib/subtract.js [hunk 4]
```diff
@@ -1,4 +1,6 @@
-const subtract = (predicate, array) => {
-
+const subtract = (num1,num2) => {
```
Comment (Moniarchy) [inline src/lib/subtract.js:1]: This could be made shorter:
` const subtract = ( num1, num2 ) => num1 - num2 `

When writing in ES6, and using arrow functions, if your function's logic is short enough to fit on one line, you may omit curly braces and there is an implicit return, so you don't even need to type out `return num1 - num2`

File: src/lib/subtract.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
...
+  var difference = num1 - num2;
+  console.log(difference);
```
Comment (Moniarchy) [inline src/lib/subtract.js:3]: remove `console.log`, there's no reason you should be logging output for this method.

File: src/spec/intersection_spec.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import chai from 'chai'
+import intersection from '../lib/intersection.js'
+const expect = chai.expect
+
+describe('Intersection', () => {
+  it('Creates an array of unique values that are included in all given arrays', () => {
+    expect(intersection([4,2,7,5],[4,3,8,6])).to.eql([4])
+  })
+  it('Creates an array of unique values that are included in all given arrays', () => {
```
Comment (Moniarchy) [inline src/spec/intersection_spec.js:9]: this test can either be dropped or merged into the previous one-- it is testing for the same thing with a different set of arrays.

File: src/spec/intersection_spec.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    expect(intersection([6,10,7],[4,3,8,6])).to.eql([6])
+  })
+  it('Creates an array of unique values that are included in all given arrays', () => {
```
Comment (Moniarchy) [inline src/spec/intersection_spec.js:12]: This unit test should have a different description that is explicit regarding what it's testing. Perhaps something like `it( 'returns an empty array when there are no matching values' )`

File: src/spec/intersection_spec.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    expect(intersection([8,10],[4,3,6])).to.eql([])
+  })
+  it('Creates an array of unique values that are included in all given arrays', () => {
```
Comment (Moniarchy) [inline src/spec/intersection_spec.js:15]: again, this is testing the same case as the first two unit tests. I would love to see tests for what happens if the values in the array are strings, objects, or other arrays. Also worth testing: what if only one array is supplied? What if there is a null value in one or both arrays?

File: src/spec/sort_spec.js [hunk 9]
```diff
@@ -1,16 +1,13 @@
 import chai from 'chai'
 import sort from '../lib/sort.js'
-import R from 'ramda'
 const expect = chai.expect
 
 describe('Sort', () => {
   it('Sorts the list in ascending order based on difference function', () => {
```
Comment (Moniarchy) [inline src/spec/sort_spec.js:6]: As long as this test is testing the right functionality and passes, cool. But I am unsure what this test is saying. Can you explain?

File: src/spec/subtract_spec.js [hunk 10]
```diff
@@ -3,8 +3,8 @@ import subtract from '../lib/subtract.js'
 import R from 'ramda'
```
Comment (Moniarchy) [inline src/spec/subtract_spec.js:3]: remove this line if you are not using this library

**Review by Moniarchy — CHANGES_REQUESTED** [review]
Right now, your test script is not finding your tests. `Warning: Could not find any test files matching pattern: dist/spec/
No test files found
npm ERR! Test failed.  See above for more details.`

This is because your test script in your package.json is set to look in a different file than where you have your tests now. Currently, you have lib and spec directories inside of a src directory, so you'll want to clean up your file structure so that your tests can be run. The src directory seems to be unnecessary. It would be great to see that you're thinking about more edge cases for your tests. I left some ideas for edge cases, and we talked about some earlier as well.

Finally, from here on out, try to make sure that you keep your git flow going. My suggestion based on the OSS projects I've contributed to here and outside of LG is that you create a branch from the master that has functionality for one method and the tests for it, do that work, push it up, submit a PR to prrr, and then create a new branch from master and repeat.
